test/integration/app-document/test/csp.js

```diff
@@ -0,0 +1,21 @@
+/* global describe, it, expect */
+
+import webdriver from 'next-webdriver'
+
+export default (context, render) => {
+  describe('With CSP enabled', () => {
+    it('should load inline script by hash', async () => {
+      const browser = await webdriver(context.appPort, '/?withCSP=hash')
+      const errLog = await browser.log('browser')
```

Comment:
This test doesn't work:

The problem here is that we are checking for `e.source === 'security'`. The logs look different, though. Here is the real output:

```js
[
  {
    level: "SEVERE",
    message:
      "http://localhost:57398/_next/static/development/dll/dll_b47375699e1691168235.js 370:0 Uncaught EvalError: Refused to evaluate a string as JavaScript because…rective: \"script-src 'self' 'nonce-test-nonce'\".\n",
    source: "javascript",
    timestamp: 1540938931907
  },
  {
    level: "SEVERE",
    message:
      "http://localhost:57398/_next/static/runtime/main.js 12550:126 Uncaught TypeError: __webpack_require__(...) is not a function",
    source: "javascript",
    timestamp: 1540938932003
  },
  {
    level: "SEVERE",
    message:
      "http://localhost:57398/favicon.ico - Failed to load resource: the server responded with a status of 404 (Not Found)",
    source: "network",
    timestamp: 1540938942074
  }
];
```

🤔